week2/class.html

```diff
@@ -0,0 +1,148 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.1.0/jquery.min.js"></script>
+        <script>
+    var names = ["Tim", "Timmy", "Timy", "Timm", "Timantha", "Timbo", "Timmey", "Timmie", "Timey", "Timie", "Timmbo", "Timi", "Timmi", "Timei", "Timmei"];
+    var students = 15;
+    var homework = 5;
+    var quiz = 3;
+    var exams = 2;
+
+
+  function table(){
+
+    var array = [];
+    var x = 0;
+    var rand = 0;
+    var name = "Tim";
+    var haverage = 0;
+    var eaverage = 0;
+    var qaverage = 0;
+    var average = 0;
+    var hgrades = [];
+    var qgrades = [];
+    var egrades = [];
+
+
+      rand = Math.floor(Math.random * 14);
+      name = names[rand];
+
+      while (x < homework){
+      rand = Math.floor(Math.random * 5);
+
+      if(rand >= 3){
+
+      rand = Math.floor(Math.random * 100);
+      rand = parseInt(rand);
+      hgrades.push(rand);
+      x++;
+
+    }
+    if (hgrades.length < 3){
+      x = x - 1;
+      }
+    }
+
+      x = 0;
+      while(x < quiz){
+        rand = Math.floor(Math.random * 3);
+
+        if(rand >= 1){
+        rand = Math.floor(Math.random * 100);
```

Comment:
this code is very similar to the code above. When you see similar code, you should pull it out into a function for re-use